### ISSUE TITLE
Expander Header not read by screen reader if it's complex

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#10282]
### Describe the bug

A screen reader is unable to read the header of an expander if the header was set to anything other than a `TextBlock`.

### Steps to reproduce the bug

1. Create an Expander where the header is overridden as such:

```xml
<muxc:Expander>
    <muxc:Expander.Header>
        <StackPanel>
            <TextBlock>First text block!</TextBlock>
            <TextBlock>Additional text!</TextBlock>
        </StackPanel>
    </muxc:Expander.Header>
</muxc:Expander>
```

2. Use a screen reader to navigate to the expander

### Expected behavior

The screen reader should read out the header content. Instead, it reads "group, collapsed"!

### Screenshots

![Image](https://github.com/user-attachments/assets/ebb63926-96f4-4289-a29f-89805b5e0919)

### NuGet package version

WinUI 2 - Microsoft.UI.Xaml 2.8.2

**Comment by carlos-zamora** [comment]
Yes, this is similar to #5820. However, I don't think that screen reader users should be expected to enter scan mode to be able to get to the content. Is the expectation that the developer sets the `AutomationProperty.Name` on the Expander directly to define the desired behavior? I can honestly see an argument either way since the header can hold any content.

**Comment by niels9001** [comment]
@carlos-zamora Is the same behavior reproducable when setting a `StackPanel` in e.g. a `Button`? If so, I think this is expected?

The default properties, like `Content` and `Header` are being used to set the `AutomationProperties.Name` it seems? So overriding this with a `StackPanel`, it basically requires the dev to set this manually? We see this all the time when setting a `StackPanel` with a `FontIcon` and `TextBlock` as the `Content` of a `Button`.
